File: selenium-python/pages/app_shell_page.py
from __future__ import annotations

from pages.base_page import BasePage


class AppShellPage(BasePage):
    nav_home = "nav-home"
    nav_settings = "nav-settings"
    nav_form = "nav-form"
    logout_btn = "logout-btn"

    def goto_home(self) -> None:
        self.click_on(self.nav_home)

    def goto_settings(self) -> None:
        self.click_on(self.nav_settings)

    def goto_form(self) -> None:
        self.click_on(self.nav_form)

    def logout(self) -> None:
        self.click_on(self.logout_btn)

File: selenium-java/src/test/java/frameworks/seleniumjava/pages/AppShellPage.java
package frameworks.seleniumjava.pages;

import org.openqa.selenium.WebDriver;
import io.qameta.allure.Step;

public final class AppShellPage extends BasePage {
  private static final String NAV_HOME = "nav-home";
  private static final String NAV_SETTINGS = "nav-settings";
  private static final String NAV_FORM = "nav-form";
  private static final String LOGOUT_BUTTON = "logout-btn";

  public AppShellPage(WebDriver driver) {
    super(driver);
  }

  @Step
  public void gotoHome() {
    clickOn(NAV_HOME);
  }

  @Step
  public void gotoSettings() {
    clickOn(NAV_SETTINGS);
  }

  @Step
  public void gotoForm() {
    clickOn(NAV_FORM);
  }

 @Step
  public void logout() {
    clickOn(LOGOUT_BUTTON);
  }
}
